pyqir-jit/src/jit.rs

```diff
@@ -24,56 +28,187 @@ pub fn run_module_file<P: AsRef<Path>>(path: P) -> Result<SemanticModel, String>
         .to_owned();
     let context_type = ContextType::File(&path_str);
     let context = BareContext::new(&ctx, context_type)?;
-    let model = run_module(&context.module)?;
-    Ok(model)
+    run_module(&context.module, entry_point)
 }
 
-pub fn run_module<'ctx>(module: &Module<'ctx>) -> Result<SemanticModel, String> {
+pub fn run_module<'ctx>(
+    module: &Module<'ctx>,
+    entry_point: Option<&str>,
+) -> Result<SemanticModel, String> {
     Target::initialize_native(&InitializationConfig::default()).unwrap();
-
     let default_triple = TargetMachine::get_default_triple();
 
     let target = Target::from_triple(&default_triple).expect("Unable to create target machine");
-
     assert!(target.has_asm_backend());
     assert!(target.has_target_machine());
 
     run_basic_passes_on(&module);
+    let entry_point = choose_entry_point(module_functions(module), entry_point)?;
 
     unsafe {
         BasicRuntimeDriver::initialize_qir_context(true);
         let _ = microsoft_quantum_qir_runtime_sys::foundation::QSharpFoundation::new();
-
         let _ = inkwell::support::load_library_permanently("");
         let execution_engine = module
             .create_jit_execution_engine(OptimizationLevel::None)
             .expect("Could not create JIT Engine");
         let simulator = Simulator::new(&module, &execution_engine);
         let main = execution_engine
-            .get_function::<unsafe extern "C" fn() -> ()>("QuantumApplication__Run")
+            .get_function::<unsafe extern "C" fn() -> ()>(&entry_point)
             .unwrap();
+
         main.call();
         Ok(simulator.get_model())
     }
 }
 
+fn choose_entry_point<'ctx>(
+    functions: impl Iterator<Item = FunctionValue<'ctx>>,
+    name: Option<&str>,
+) -> Result<String, String> {
+    let mut entry_points = functions
+        .filter(is_entry_point)
+        .map(|f| f.get_name().to_str().unwrap().to_owned())
```

Comment:
Should we use expect here instead of unwap?

Reply:
I refactored this: instead of unwrapping the function name, I wrap the provided entry point name in `Ok` when comparing. If converting the CStr to a str fails, then the names can't match so we can ignore it.